core/src/main/java/com/linecorp/armeria/client/cookie/DefaultCookieJar.java

```diff
@@ -177,7 +189,7 @@ private void filterGet(Set<Cookie> cookies, String host, String path, boolean se
                         it.remove();
                         break;
                     }
-                    if (isExpired(cookie)) {
+                    if (state(cookie) == CookieState.EXPIRED) {
```

Comment:
Can we introduce an optimized version of `state()` here?

- We could avoid lock re-entrance
- We don't really care about distinguishing whether the cookie exists or not.